web/mysql_typemap.go

```diff
@@ -0,0 +1,263 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package web
+
+import (
+	"github.com/cloudspannerecosystem/harbourbridge/internal"
+	"github.com/cloudspannerecosystem/harbourbridge/spanner/ddl"
+)
+
+// TODO(searce): Decide on the issue message that needs to be shown
+// in frontend.
+var mysqlTypeMap = map[string][]typeIssue{
+	"bool": []typeIssue{
+		typeIssue{T: ddl.Bool},
+		typeIssue{T: ddl.String, Issue: internal.Widened},
+		typeIssue{T: ddl.Int64, Issue: internal.Widened}},
+	"boolean": []typeIssue{
+		typeIssue{T: ddl.Bool},
+		typeIssue{T: ddl.String, Issue: internal.Widened},
+		typeIssue{T: ddl.Int64, Issue: internal.Widened}},
+	"varchar": []typeIssue{
+		typeIssue{T: ddl.String},
+		typeIssue{T: ddl.Bytes, Issue: internal.Widened, Brief: internal.IssueDB[internal.Widened].Brief}},
```

Comment:
Why put the brief in for this one, and not the others?
In face, why have brief here at all -- can't we just look up up using IssueDB?

Reply:
Brief is used in frontend to show warning message when user selects different type than default type, i'll remove Issue field from typeMap as it is not required in frontend anymore.

And the reason why we use different message in brief is that in some type it may require a different explanation than in IssueDB. for example set or enum type.
Alternatively we can just map to NoGoodType issue from IssueDB, what do you think?

Reply:
We not use IssueDB as source of briefs -- code is now in web/web.go (see addTypeToList).